thriftpy/contrib/aio/client.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+import asyncio
+import functools
+from thriftpy.thrift import args2kwargs
+from thriftpy.thrift import TApplicationException, TMessageType
+
+
+class TAsyncClient:
+
+    def __init__(self, service, iprot, oprot=None):
+        self._service = service
+        self._iprot = self._oprot = iprot
+        if oprot is not None:
+            self._oprot = oprot
+        self._seqid = 0
+
+    def __getattr__(self, _api):
+        if _api in self._service.thrift_services:
+            return functools.partial(self._req, _api)
+
+        raise AttributeError("{} instance has no attribute '{}'".format(
+            self.__class__.__name__, _api))
+
+    def __dir__(self):
+        return self._service.thrift_services
+
+    @asyncio.coroutine
+    def _req(self, _api, *args, **kwargs):
+        _kw = args2kwargs(getattr(self._service, _api + "_args").thrift_spec,
+                          *args)
+        kwargs.update(_kw)
+        result_cls = getattr(self._service, _api + "_result")
+
+        yield from self._send(_api, **kwargs)
+        # wait result only if non-oneway
+        if not getattr(result_cls, "oneway"):
+            return (yield from self._recv(_api))
+
+    @asyncio.coroutine
+    def _send(self, _api, **kwargs):
+        self._oprot.write_message_begin(_api, TMessageType.CALL, self._seqid)
+        args = getattr(self._service, _api + "_args")()
```

Comment:
`thriftpy` simply ignored `seqid`, but we better fix it.

Reply:
I think it's not a problem about asyncio support, maybe we can create another pull request to solve it both sync and async module.